xonsh_jupyter/shell.py

```diff
@@ -3,7 +3,12 @@
 import io
 import sys
 
-from xonsh.base_shell import BaseShell
+try:
+    # xonsh 0.18.0+
+    from xonsh.shells.base_shell import BaseShell
+except ImportError:
+    from xonsh.base_shell import BaseShell
+
```

Comment:
it is better to require the xonsh version in pyproject.toml and be released as a new version